### PR TITLE
Reveal identity polishing

### PR DESCRIPTION
**Pull request by SamTremko**

### CODE REVIEW

**Comment by SamTremko** [comment]
* **#1108** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1108?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1105** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1105?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1098** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1098?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1097** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1097?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1094** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1094?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1093** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1093?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1092** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1092?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1089** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1089?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1088** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1088?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1087** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1087?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @SamTremko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/mobile/src/components/ChatDetailScreen/components/MessagesList.tsx [hunk 1]
```diff
@@ -15,16 +25,29 @@ function renderItem({
...
   return (
     <FlashList
       data={dataAtoms}
       estimatedItemSize={54}
+      contentContainerStyle={contentStyle.contentContainerStyle}
       keyExtractor={atomKeyExtractor}
       inverted
       renderItem={renderItem}
+      onViewableItemsChanged={
+        handleIsRevealIdentityOrContactRevealMessageVisible
```
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/components/MessagesList.tsx:46]: Debounce - when user has a lot of messages + user has lot of contacts there might be a big performance impact

File: apps/mobile/src/components/ChatDetailScreen/components/MessagesList.tsx [hunk 2]
```diff
@@ -15,16 +25,29 @@ function renderItem({
...
+      }
+      viewabilityConfig={{
+        itemVisiblePercentThreshold: LIST_ITEM_VISIBILITY_PERCENTAGE_THRESHOLD,
```
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/components/MessagesList.tsx]: Move object creation out of react component just to be sure this does not cause any unnecessary rerenders

File: apps/mobile/src/components/ChatDetailScreen/components/MessagesList.tsx [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
-import {useAtomValue, type Atom} from 'jotai'
+import {useAtomValue, useSetAtom, type Atom} from 'jotai'
 import React from 'react'
+import {StyleSheet} from 'react-native'
 import atomKeyExtractor from '../../../utils/atomUtils/atomKeyExtractor'
 import {chatMolecule} from '../atoms'
 import MessageItem, {type MessagesListItem} from './MessageItem'
 
+const LIST_ITEM_VISIBILITY_PERCENTAGE_THRESHOLD = 30
+const QUICK_ACTION_BANNER_HEIGHT_WITH_PADDING = 80
```
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/components/MessagesList.tsx]: Use the same constant for setting height of quick action banner so if we decide to change height of quick action banner we don't break UX

File: apps/mobile/src/state/chat/utils/processIdentityRevealMessageIfAny.ts [hunk 4]
```diff
@@ -35,13 +38,15 @@ export default function processIdentityRevealMessageIfAny(
...
             type: 'svgXml',
-            svgXml: avatarsSvg[
-              randomNumberFromSeed(
-                0,
-                avatarsSvg.length - 1,
-                identityRevealMessage.message.deanonymizedUser.name
-              )
-            ] as SvgString,
+            svgXml:
+              chat.chat.otherSide.realLifeInfo?.anonymousImage ??
```
Comment (kaladivo) [inline apps/mobile/src/state/chat/utils/processIdentityRevealMessageIfAny.ts]: Do not save the image to chat state - use the function to generate the same image for pubic key

File: apps/mobile/src/components/ChatDetailScreen/atoms/index.tsx [hunk 5]
```diff
@@ -104,6 +105,126 @@ export const chatMolecule = molecule((getMolecule, getScope) => {
...
+          message.type === 'message' &&
+          message.message.state === 'received' &&
+          (message.message.message.messageType === 'REQUEST_REVEAL' ||
+            message.message.message.tradeChecklistUpdate?.identity?.status ===
+              'REQUEST_REVEAL')
+      )
+    }
+  )
+
+  const isRevealIdentityMessageVisibleAtom = atom(true)
```
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/atoms/index.tsx]: isRevealIdentityMessageDismissedAtom

File: apps/mobile/src/components/ChatDetailScreen/atoms/index.tsx [hunk 6]
```diff
@@ -104,6 +105,126 @@ export const chatMolecule = molecule((getMolecule, getScope) => {
...
+          message.type === 'message' &&
+          message.message.state === 'received' &&
+          (message.message.message.messageType === 'REQUEST_CONTACT_REVEAL' ||
+            message.message.message.tradeChecklistUpdate?.contact?.status ===
+              'REQUEST_REVEAL')
+      )
+    }
+  )
+
+  const isContactRevealMessageVisibleAtom = atom(true)
```
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/atoms/index.tsx]: isContactRevealMessageDismissedAtom

**Review by kaladivo — APPROVED** [review]

**Pull request merged**